source/common/common/posix/thread_impl.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <functional>
+
+#include "envoy/thread/thread.h"
+
+namespace Envoy {
+namespace Thread {
+
+class ThreadIdImplPosix : public ThreadId {
+public:
+  ThreadIdImplPosix(int32_t id);
+
+  std::string debugString() const override;
+
```

Comment:
Nit: prefer for the `override`s that there is no space between them and we start the block with something like `// ThreadId overrides.`